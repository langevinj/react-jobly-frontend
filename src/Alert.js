import React from 'react'

//return and alert div for each message received
function Alert({ type, messages }){
    return (
        <>
        {messages.map(message => <div className={`alert alert-${type}`} key={message}>{message}</div>)}
        </>
    )
}

Alert.defaultProps = {
    type: "danger",
    messages: []
}

export default Alert;