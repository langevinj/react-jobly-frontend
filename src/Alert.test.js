import React from 'react'
import { render } from '@testing-library/react'
import Alert from './Alert'

it("renders without crashing", function(){
    render(<Alert />);
});

it("matches the snapshot for danger", function() {
    let messages = ["Invalid entry", "Database erased"];
    const { asFragment } = render(<Alert type="danger" messages={messages} />)
    expect(asFragment()).toMatchSnapshot();
});

it("matches the snapshot for success", function() {
    let messages = ["Login Successful"]
    const { asFragment } = render(<Alert type="success" messages={messages}/>)
    expect(asFragment()).toMatchSnapshot();
})