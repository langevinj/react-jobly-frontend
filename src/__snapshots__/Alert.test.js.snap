// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot for danger 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
  >
    Invalid entry
  </div>
  <div
    class="alert alert-danger"
  >
    Database erased
  </div>
</DocumentFragment>
`;

exports[`matches the snapshot for success 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Login Successful
  </div>
</DocumentFragment>
`;
